<?xml version="1.0"?>

<launch>
  
  <node pkg="vehicle_controller" type="controller_node" name="vehicle_controller" output="screen" respawn="true">
    
    <!--<remap from="/cmd_vel_raw" to="/cmd_vel_remap" />-->
    <remap from="/state" to="/odom" />
    
    <param name="frame_id" value="world" />

    <param name="pub_cmd_hz" value="10" />
    <param name="rot_correction_factor" value="1.0" />
    <param name="k_p_rotation" value="5" />
    <param name="execution_period" value="0.5" />
    <param name="update_skip_until_vel_increase" value="5" />
    <param name="lower_al_angle" value="0.0" />
    <param name="upper_al_angle" value="1.0" />
    <param name="enable_angle_compensation" value="true" />
    <param name="enable_ground_compensation" value="true" />
    <param name="enable_velocity_increase" value="true" />
    <param name="show_trajectory_planing" value="false" />
    <param name="stability_angle" value="1.0" />

    <param name="carrot_distance" value="0.4" />
    <param name="min_speed" value="0.0" />
    <param name="velocity_blocked_limit" value="0.6" />
    <param name="velocity_blocked_time" value="3.0" />
    
    <param name="y_symmetry" value="false" />
    
    <param name="max_controller_speed" value="0.8" />
    <param name="max_unlimited_speed" value="0.6" />
    <param name="max_unlimited_angular_rate" value="3.0" />
    <param name="max_controller_angular_rate" value="1.4" />
    
    <param name="vehicle_type" value="tracked" />
    <param name="vehicle_control_type" value="differential_steering"/>
    <param name="pd_params" value="PdParamsArgo" />

    <param name="goal_position_tolerance" value="0.1" />
    <param name="goal_angle_tolerance" value="0.05" />
    <param name="speed" value="0.6" />
    
    <param name="wheel_separation" value="0.34" />
    
    <param name="check_stuck" value="true" />
    
    <!--<remap from="/cmd_vel_teleop" to="/cmd_vel_teleop_safe" />-->

  </node>
  
</launch>

